test/e2e_federation/replicaset.go

```diff
@@ -369,6 +369,11 @@ func createReplicaSetOrFail(clientset *fedclientset.Clientset, replicaset *v1bet
 	}
 	By(fmt.Sprintf("Creating federation replicaset %q in namespace %q", replicaset.Name, namespace))
 
+	// a hack to pass validation check for controller selector immutability prior create (technically an update to nil object)
+	consistentLabelMap := map[string]string{"name": "myrs-consistent"}
+	replicaset.Spec.Selector.MatchLabels = consistentLabelMap
+	replicaset.Spec.Template.ObjectMeta.Labels = consistentLabelMap
```

Comment:
It seems that the validation check will break some tests. I still need to sync with @kow3ns what is the best way to add the validation without breaking any test.